src/bridges/convex/ConvexStakingBridge.sol

```diff
@@ -0,0 +1,195 @@
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import {BridgeBase} from "../base/BridgeBase.sol";
+import {AztecTypes} from "../../aztec/libraries/AztecTypes.sol";
+import {ICurvePool} from "../../interfaces/curve/ICurvePool.sol";
+import {ErrorLib} from "../base/ErrorLib.sol";
+import {IRollupProcessor} from "../../aztec/interfaces/IRollupProcessor.sol";
+import {IConvexFinanceBooster} from "../../interfaces/convex/IConvexFinanceBooster.sol";
+import {IConvexToken} from "../../interfaces/convex/IConvexToken.sol";
+import {ICurveLpToken} from "../../interfaces/convex/ICurveLpToken.sol";
+import {ICurveRewards} from "../../interfaces/convex/ICurveRewards.sol";
+
+
+/**
+ * @notice A DefiBridge that stakes Curve LP Token into Convex Finance. Unstaking supported as well.
+ * @notice A DefiBridge that stakes Curve LP Token into Convex Finance. Convex Finance transfers Curve LP Tokens 
+ * from bridge to a liquidity gauge. New Convex Tokens are minted and then staked (transfered) to CRV Rewards.
+ * Bridge mints a matching version of the staked Convex Tokens, CSB Tokens, and assignes their ownership to the RollUpProcessor.
+ * At unstaking, both - CSB Tokens and Convex Tokens - are burned and ownership of Curve LP Tokens is given back to the RollUpProcessor.
+ * @dev Synchronous and stateless bridge that will hold no funds beyond dust for gas-savings.
+ * @author Vojtech Kaiser
+ */
+contract ConvexStakingBridge is BridgeBase, ERC20("ConvexStakingBridge", "CSB") {
+    using SafeERC20 for IConvexFinanceBooster;
+    using SafeERC20 for IConvexToken;
+    using SafeERC20 for ICurveLpToken;
+    // using SafeERC20 for ICurveRewards; // not really needed...I am not sending anything, just checking balance
+
+    // CONTRACTS
+    // Deposit Contract for Convex Finance - Main File
+    IConvexFinanceBooster public constant CONVEX_DEPOSIT = IConvexFinanceBooster(0xF403C135812408BFbE8713b5A23a04b3D48AAE31);
+    // General Interface for any Curve LP Token Contract
+    ICurveLpToken public CURVE_LP_TOKEN;
+    // General Interface for any Curve Rewards Contract
+    ICurveRewards public CURVE_REWARDS;
+    
+    // POOLS
+    uint public lastPoolLength;
+
+    struct PoolInfo {
+        uint poolPid;
+        address curveLpToken;
+        address curveRewards;
+        bool exists;
+    }
+    mapping(address => PoolInfo) public pools;
+
+    mapping(uint => bool) public invalidPoolPids;
+
+    // ERRORS
+    error invalidTotalInputValue();
+    error invalidAssetType();
+    error invalidInputOutputAssets();
+    error invalidPoolPid();
+
+    constructor(address _rollupProcessor) BridgeBase(_rollupProcessor) {
+        addInvalidPoolPid(48);
+        // _mint(address(this), DUST) // for optimization
+    }
+
+    /** 
+    @notice Add a new pool pid that should not be allowed to interact with.
+    @param poolPid Id of a pool to add.
+    */
+    function addInvalidPoolPid(uint poolPid) public onlyRollup {
```

Comment:
The bridge doesn't manage to get deployed in the E2E test because you use the onlyRollup modifier in this function which gets called from within constructor and reverts. In the unit tests, the test contract itself is passed to the bridge as the rollupProcessor and for this reason the deployment succeeds there.